src/mca/gds/shmem/gds_shmem.c

```diff
@@ -1978,9 +2001,31 @@ server_add_nspace(
     pmix_info_t info[],
     size_t ninfo
 ) {
-    PMIX_HIDE_UNUSED_PARAMS(nspace, nlocalprocs, info, ninfo);
+    PMIX_HIDE_UNUSED_PARAMS(nlocalprocs);
     PMIX_GDS_SHMEM_VVOUT_HERE();
-    return PMIX_SUCCESS;
+
+    // Create a job tracker for this nspace.
+    pmix_gds_shmem_job_t *job;
+    pmix_status_t rc = pmix_gds_shmem_get_job_tracker(nspace, true, &job);
+    if (PMIX_UNLIKELY(PMIX_SUCCESS != rc)) {
+        PMIX_ERROR_LOG(rc);
+        return rc;
+    }
+
+    for (size_t i = 0; i < ninfo; ++i) {
+        if (PMIX_CHECK_KEY(&info[i], PMIX_USERID)) {
+            const uid_t nuid = (uid_t)info[i].value.data.uint32;
+            PMIX_GDS_SHMEM_VOUT(
+                "%s: updating nspace=%s UID from %zd to %zd",
+                __func__, nspace, (size_t)job->uid, (size_t)nuid
+            );
+            job->uid = nuid;
+            job->chown = true;
+            // We don't care about any other keys, so bail.
```

Comment:
You probably should check for `PMIX_GRPID` as well and update the group permissions too if it doesn't match the value returned by `getegid`. Many sites use the group ID as an accounting entity and we should allow for sharing between members of the group.